lesson_9.py

```diff
@@ -0,0 +1,180 @@
+            self.__num_employees = new_num_employees
+            print(f'Number of employees updated to {new_num_employees}')
+        else:
+            raise ValueError(f'Exceeded max capacity of {AppleCompany.__MAX_EMPLOYEES_CAPACITY}')
+
+    @property
+    def company_location(self):
+        """Getter method for the private attribute __location."""
+        return self.__location
+
+    def get_profit_by_location(self, revenue: int, expenses: int):
+        """
+        Calculate the profit in the company's location.
+        Args: - revenue (int): The revenue of the company.
+              - expenses (int): The expenses of the company.
+        """
+        profit = revenue - expenses
+        print(f'Profit in {self.__location} is {profit} USD')
+
+    @classmethod
+    def from_country(cls, country: str):
+        """
+        Class method to create an instance of AppleCompany based on the country.
+        Args: - country (str): The country to get location of the company, taking result by key's value
+        from offices_locations dictionary
+        Returns: An instance of the AppleCompany class.
+        """
+        location = AppleCompany.__offices_locations.get(country, 'Unknown')  # 'Unknown' - default value
+        return cls(location)
+
+    @staticmethod
+    def get_production_cost(production_type: str):
+        """
+        Static method to get the production cost of a specific Apple product.
+        Args: - production_type (str): The type of Apple product to get the production cost for.
+        Returns:- str: A string that displays the production cost for the given product type.
+        """
+        production_cost = 0
+        if production_type == 'iPhone':
+            production_cost = 500
+        elif production_type == 'iPad':
+            production_cost = 350
+        elif production_type == 'MacBook':
+            production_cost = 1000
+        else:
+            print('Invalid production type')
+        return f'{production_type} cost is {production_cost} USD'
+
+
+# Create a class with a description of the worker. Any worker, employees.
+class Employees:
+
+    def __init__(self, name, age):
+        self.__name = name
+        self.__age = age
+        self.__position = ''
+        self.__min_salary = 2000
+
+    @property
+    def position(self):
+        return self.__position
+
+    @property
+    def min_salary(self):
+        return self.__min_salary
+
+    @property
+    def age(self):
+        return self.__age
+
+    @age.setter
+    def age(self, new_age: int):
+        """
+        Sets the age of the Employee to a new value.
+        param new_age: int
+        """
+        if isinstance(new_age, int):
+            if new_age < self.__age:
+                raise ValueError(f'Sorry, but employee cant get younger. Re-check age value')
+            elif new_age >= self.__age:
+                self.__age = new_age
+
+    def employee_apply_for_job(self, position, min_salary):
```

Comment:
it is good practice to start function naming from verb. 
Also you do not need specify employee_ here because it is already related to class employee